channel/src/main/java/com/synopsys/integration/alert/channel/jira2/cloud/delegate/JiraCloudIssueTransitioner.java

```diff
@@ -54,85 +33,27 @@ public JiraCloudIssueTransitioner(
-        StatusCategory issueStatusCategory = retrieveIssueStatusCategory(existingIssueDetails.getIssueKey());
-        String statusCategoryKey = issueStatusCategory.getKey();
-        if (IssueOperation.OPEN.equals(issueTransitionType)) {
-            // Should reopen?
-            return DONE_STATUS_CATEGORY_KEY.equals(statusCategoryKey);
-        } else if (IssueOperation.RESOLVE.equals(issueTransitionType)) {
-            // Should resolve?
-            return TODO_STATUS_CATEGORY_KEY.equals(statusCategoryKey);
-        }
-        return false;
-    }
-
-    private StatusCategory retrieveIssueStatusCategory(String issueKey) throws AlertException {
-        try {
-            StatusDetailsComponent issueStatus = issueService.getStatus(issueKey);
-            return issueStatus.getStatusCategory();
-        } catch (IntegrationException e) {
-            throw new AlertException(String.format("Failed to retrieve issue status from Jira. Issue Key: %s", issueKey), e);
-        }
+    protected String extractResolveTransitionName() {
+        return distributionDetails.getResolveTransition();
     }
 
     @Override
-    protected void findAndPerformTransition(ExistingIssueDetails<String> existingIssueDetails, String transitionName) throws AlertException {
-        String issueKey = existingIssueDetails.getIssueKey();
-        logger.debug("Attempting the transition '{}' on the issue '{}'", transitionName, issueKey);
-        List<TransitionComponent> issueTransitions = retrieveTransitions(issueKey);
-        Optional<IdComponent> foundTransitionId = findTransitionIdByTransitionName(issueTransitions, transitionName);
-        if (foundTransitionId.isPresent()) {
-            performTransition(issueKey, foundTransitionId.get());
-        } else {
-            List<String> validTransitions = gatherValidTransitionNames(issueTransitions);
-            throw new IssueMissingTransitionException(issueKey, transitionName, validTransitions);
-        }
-    }
-
-    private List<TransitionComponent> retrieveTransitions(String issueKey) throws AlertException {
-        try {
-            TransitionsResponseModel transitionsResponse = issueService.getTransitions(issueKey);
-            return transitionsResponse.getTransitions();
-        } catch (IntegrationException e) {
-            throw new AlertException(String.format("Failed to retrieve transitions from Jira. Issue Key: %s", issueKey), e);
-        }
-    }
-
-    private Optional<IdComponent> findTransitionIdByTransitionName(List<TransitionComponent> transitions, String transitionName) {
-        return transitions
-                   .stream()
-                   .filter(transitionComp -> transitionComp.getName().equals(transitionName))
-                   .map(TransitionComponent::getId)
-                   .map(IdComponent::new)
-                   .findFirst();
+    protected StatusDetailsComponent getIssueStatus(String issueKey) throws IntegrationException {
```

Comment:
I forget what we decided last time. Do we prefer `fetch` or `retrieve` as a prefix?

Reply:
It was fetch. I'll refactor these